src/parser/md.rs

```diff
@@ -149,53 +157,34 @@ impl Default for MdParser {
 }
 
 impl MdParser {
-    const LINK_PATTERN: &'static str = r"\[([^\[\]]*)\]\((.*?)\)";
-
-    /// Check the validity of the parser configuration
-    pub fn check(&self) -> Result<(), String> {
-        Ok(())
-    }
+    /// Sets the default language of the returned parser (or does nothing if `None` is passed)
+    pub fn default_language(&self, language: Option<String>) -> Self {
+        let mut cloned = self.clone();
 
-    /// Creates a default parser with a fallback language
-    pub fn for_language(language: String) -> Self {
-        Self {
-            default_language: Some(language),
-            ..Self::default()
+        if language.is_some() {
+            cloned.default_language = language;
         }
-    }
 
-    /// Sets the default language of this parser (or does nothing if `None` is passed)
-    pub fn default_language(&self, language: Option<String>) -> Self {
-        if let Some(language) = language {
-            Self {
-                default_language: Some(language),
-                ..self.clone()
-            }
-        } else {
-            self.clone()
-        }
+        cloned
     }
 
     fn parse_transclusion(&self, line: &str, into: &Path) -> Fallible<Option<Node>> {
-        let trim = line.trim();
-        if trim.starts_with(&self.transclusion_start) {
-            if let Some(index) = line.find(&self.transclusion_end) {
-                let trans = &trim[self.transclusion_start.len()..index];
-                let regex = Regex::new(Self::LINK_PATTERN).unwrap();
```

Comment:
Oh, I see that this was quite an evil line with a lot of unnecessary overhead.

Reply:
I guess it depends: The regular expression was only compiled when the transclusion prefix is found, so it is not done for every single line. But of course, generally speaking one should not try to compile regular expression as seldomly as possible which in this case of a constant pattern is once.

(I think this snippet did actually contain a bug though: `index` refers to the full `line` but is used to index into `trim`, i.e. if there is a lot of whitespace at the beginning and `line` is much longer than `trim`, this indexing could actually panic.)